core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -211,6 +224,59 @@ private void runWithCompose(String cmd) {
                 .invoke();
     }
 
+    @SuppressWarnings("unchecked")
+    private static void validate(File composeFile) {
+        Yaml yaml = new Yaml();
+        try (FileInputStream fileInputStream = FileUtils.openInputStream(composeFile)) {
+            Object template = yaml.load(fileInputStream);
+            validate(template, composeFile.getAbsolutePath());
+        } catch (IOException e) {
+            log.warn("Failed to read YAML from {}", composeFile.getAbsolutePath(), e);
+        }
+    }
+
+    @VisibleForTesting
+    static void validate(Object template, String identifier) {
+        if (!(template instanceof Map)) {
```

Comment:
I think we could extract 240-261 into a private function and then have 
```
if (!hasKnownFormat(object)) {
  log.debug("Compose file {} has an unknown format", identifier);
  return
}
...
```

Reply:
extracted methods make it much harder to follow the structure IMO, especially when dealing with the format/structure validation

Reply:
I disagree, since it makes the responsibility of the part of the code clearer (therefore also increasing maintainability).
But we don't need to fight over this :wink: